File: backend/app/models/terms.py
from datetime import datetime
from pydantic import BaseModel


class Terms(BaseModel):
    id: int
    version: str
    content: str
    created_at: datetime
    archived: bool = False


class TermsAcceptance(BaseModel):
    user_id: str
    account_type: str
    version: str
    accepted_at: datetime
    accepted_ip: str
    user_agent: str

File: lib/terms.js
import { TERMS } from '../data/terms'
import { getStore, addAuditLog } from './memory-store'

export function getLatestTerms() {
  return TERMS[TERMS.length - 1]
}

export function recordAcceptance({ userId, accountType, version, ipAddress, userAgent }) {
  const store = getStore()
  const entry = {
    userId,
    accountType,
    version,
    ipAddress,
    userAgent,
    acceptedAt: new Date().toISOString()
  }
  store.termsAcceptances.push(entry)
  addAuditLog({
    event: 'terms.accepted',
    userId,
    accountType,
    version,
    ipAddress,
    userAgent
  })
  return entry
}
